### ISSUE TITLE
Crash when pressing Add Button

### ISSUE THREAD

**Issue by dotswift** [derekquinn/ios-tasks#1]
Possible cause:`saveTask()`in ViewController.swift

**Comment by JorgeCurioso** [comment]
You're on the right track! You had the right approach to reading & writing values from `UserDefaults` 💯 , but it has a few nuances that are goofing things up for ya
- The typical way to interact with `UserDefaults` is to reference a shared instance called `standard`, which is the basic bucket of key/value storage allocated to your app (stored in your app's "bundle") accessible each time you run your app.
- The difference between `Class`/`Type` methods & `instance` methods will dictate which methods are available, and sometimes there are class & instance methods on the same type which confuse things. 
- Here, you want to use the instance methods of `integer(forKey:)` (or `value(forKey:)`) & `set(_ forKey:)`), so calling them on the `standard` instance will work ok

```swift
// the integer method will return the Int type here as a non-optional
// but using value(forKey:) as? Int works fine as well
let count = UserDefaults.standard.integer(forKey:"count")

let newCount = count + 1
        
UserDefaults.standard.set(newCount, forKey: "count")
UserDefaults.standard.set(text, forKey: "task_\(newCount)")
```

you can also create a reference in the file, or locally in the method as well, & do the same
```swift
let defaults = UserDefaults.standard

defaults.set(newCount, forKey: "count") 
// etc.
```

Hope that helps - app is lookin' good!

**Comment by dotswift** [comment]
@JorgeCurioso thanks for the kind assist. I have another session planned on Sunday and will address this issue then.

**Comment by dotswift** [comment]
@JorgeCurioso Marking this issue resolved, the latest commit `eb50157` addresses it nicely thanks to your help.
